app/src/main/java/ch/epfl/skysync/ui/components/FlightForm.kt

```diff
@@ -0,0 +1,666 @@
+                        IconButton(
+                            modifier =
+                                Modifier.padding(horizontal = defaultPadding)
+                                    .testTag("Add Vehicle Button"),
+                            onClick = {
+                              if (listVehiclesValue.size < allVehicles.size) {
+                                addVehicle = true
+                              }
+                            },
+                        ) {
+                          Icon(Icons.Default.Add, contentDescription = "Add Vehicle")
+                        }
+                      }
+                }
+                listVehiclesValue.withIndex().forEach() { (idList, car) ->
+                  item {
+                    var expandedVehicleMenu by remember { mutableStateOf(false) }
+                    ExposedDropdownMenuBox(
+                        modifier =
+                            Modifier.fillMaxWidth()
+                                .clickable { expandedVehicleMenu = true }
+                                .testTag("Vehicle Menu $idList"),
+                        expanded = expandedVehicleMenu,
+                        onExpandedChange = { expandedVehicleMenu = !expandedVehicleMenu }) {
+                          Row(
+                              modifier = Modifier.fillMaxWidth(),
+                              horizontalArrangement = Arrangement.SpaceBetween,
+                              verticalAlignment = Alignment.CenterVertically) {
+                                MenuDropDown(
+                                    defaultPadding = defaultPadding,
+                                    title = "Vehicle",
+                                    value = car,
+                                    onclickMenu = { item -> listVehiclesValue[idList] = item },
+                                    items = allVehicles,
+                                    showString = { it.name })
+                                IconButton(
+                                    modifier = Modifier.testTag("Delete Vehicle Button"),
+                                    onClick = {
+                                      listVehiclesValue.removeAt(idList)
+                                      if (listVehiclesValue.isEmpty()) {
+                                        addVehicle = true
+                                      }
+                                    },
+                                ) {
+                                  Icon(
+                                      Icons.Default.Delete,
+                                      contentDescription = "Delete Vehicle $idList")
+                                }
+                              }
+                        }
+                  }
+                }
+                if (addVehicle) {
+                  item {
+                    var expandedVehicleMenu by remember { mutableStateOf(false) }
+
+                    ExposedDropdownMenuBox(
```

Comment:
Is there a specific reason you don't use your `MenuDropDown` component here ?

Reply:
Not really, I have changed it. But as a side effect the TestTags of this field have also changed